src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheck.java

```diff
@@ -57,6 +57,55 @@
  * <pre>
  * &lt;module name=&quot;InnerAssignment"/&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * class MyClass {
+ *
+ *   void foo() {
+ *     int a, b;
+ *     a = b = 5; // violation; assignment to each variable should be in a separate statement
+ *
+ *     a = 5; // OK
+ *     b = 5; // OK
+ *     a = 5; b = 5; // OK
+ *
+ *     double myDouble;
+ *     double[] doubleArray = new double[] {myDouble = 4.5, 15.5}; // violation
+ *     // assignment to 'myDouble' should be in a separate statement
+ *
+ *     String nameOne;
+ *     List&lt;String&gt; myList = new ArrayList&lt;String&gt;();
+ *     myList.add(nameOne = "tom"); // violation
+ *     // assignment to 'nameOne' should be in a separate statement
+
+ *     for (int k = 0; k &lt; 10; k = k + 2) { // OK
+ *       // some code
+ *     }
+ *
+ *     boolean someVal;
+ *     if (someVal = true) { // violation, assignment to `someVal` should be in a separate statement
+ *       // some code
+ *     }
+ *
+ *     while (someVal = false) {} // violation, assignment to `someVal` should be in separate statement
+ *     while ((someVal = false) {} // violation, assignment to `someVal` should be in a separate statement
+ *
+ *     InputStream is = new FileInputStream("textFile.txt");
+ *     while ((b = is.read()) != -1) { // OK, this is a common idiom
+ *       // work with b
+ *     }
+ *
+ *     a = (someVal = false) ? 5 : 10; // violation, assignment to `someVal` should be in a separate statement
+ *     a = someVal ? (b = 5) : 10; // violation, assignment to `b` should be in a separate statement
+ *     a = someVal ? 5 : b += 10; // violation, assignment to `b` should be in a separate statement
```

Comment:
`b += 10;` please make it a separate case, no need for ternary operator

Reply:
Removed. I've create a separate example for `b += 10`